src/services/request-service.ts

```diff
@@ -0,0 +1,75 @@
+import Promise = require('bluebird');
+import r = require('rethinkdb');
+import uuid = require('uuid');
+import EmailService = require('./email-service');
+import nodemailer = require('nodemailer');
+import q = require('../dbutils/query');
+import models = require('../models/models');
+import errors = require('../errors/errors');
+import config = require('../config');
+
+
+module RequestService {
```

Comment:
There are some minor race condition bugs in this file. For example, in `createRequest()` method, you call `requestExists()` method. There can be a case where `requestExists()` return `false` but before `createRequest()` insert the new request, another "request" will be added to the database.

So basically by the time `requestExists()` return, the state could change and invalidate the result of `requestExists()`. I think it would be better to refactor out the `reuquestExists` query and  make the db calls in `createRequest()`, `removeRequest()` into a single query with a branch. Take a look here for an example: https://github.com/FroYoInc/wafflecone/blob/develop/src/services/carpool.svc.ts#L82